flytepropeller/pkg/controller/nodes/catalog/datacatalog/datacatalog.go

```diff
@@ -251,20 +251,16 @@ func (m *CatalogClient) CreateArtifact(ctx context.Context, key catalog.Key, dat
 	}
 	_, err = m.client.AddTag(ctx, &datacatalog.AddTagRequest{Tag: tag})
 	if err != nil {
-		if status.Code(err) == codes.AlreadyExists {
-			logger.Warnf(ctx, "Tag %v already exists for Artifact %v (idempotent)", tagName, cachedArtifact.Id)
-		} else {
-			logger.Errorf(ctx, "Failed to add tag %+v for artifact %+v, err: %+v", tagName, cachedArtifact.Id, err)
-			return catalog.Status{}, err
-		}
+		logger.Errorf(ctx, "Failed to add tag %+v for artifact %+v, err: %+v", tagName, cachedArtifact.Id, err)
+		return catalog.Status{}, err
```

Comment:
Love the rest of it. Can we leave this as an idempotent operation and not return an error (ie. `PUT_FAILURE`) on already exists? Unless we add a `reason` that says `Artifact already exists` I feel like users are going to see a failure here and think something broke.

Reply:
yup sounds good. I think the easiest approach would be similar to adding a new cache_status ~ALREADY_EXISTS. 

I'll make note to add that when I get back into the console, IDL sync issue for the evict cache status.